### PR TITLE
Fix docker auth error during template build

### PR DESCRIPTION
**Pull request by jakubno**
# Description

For some users docker client isn't sending `Authorization` header, this is a temporary workaround until we resolve the issue directly with Docker.

It uses a local proxy, which handles the authentication.

This fixes: #450

### CODE REVIEW

**Review by mlejva — APPROVED** [review]

File: apps/web/src/app/(docs)/docs/cli/build-auth-error/page.mdx [hunk 1]
```diff
@@ -0,0 +1,70 @@
+# Docker push authentication error
+
+When you try to push a Docker image to a registry, you might encounter an authentication error. This error sometimes occurs for some users, Docker doesn't send any credentials to the registry.
```
Comment (ValentaTomas) [inline apps/web/src/app/(docs)/docs/cli/build-auth-error/page.mdx]:
```suggestion
When the CLI tries to push a Docker image to the registry, you might encounter an authentication error. This error sometimes occurs for users when Docker doesn't send any credentials to the registry.
```

File: packages/cli/src/commands/template/buildWithProxy.ts [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+export async function buildWithProxy(
+  userConfig: UserConfig | null,
+  connectionConfig: e2b.ConnectionConfig,
+  accessToken: string,
+  template: { templateID: string; buildID: string },
+  root: string
+) {
+  if (!userConfig?.dockerProxySet) {
+    console.log(
+      "There was an issue with Docker authentication. Here's a temporary workaround: https://e2b.dev/docs/cli/build-auth-error. Once you've followed the guide, please continue."
```
Comment (ValentaTomas) [inline packages/cli/src/commands/template/buildWithProxy.ts]:
```suggestion
      "There was an issue during Docker authentication. Please follow the workaround steps from https://e2b.dev/docs/cli/build-auth-error and then continue."
```

File: apps/web/src/app/(docs)/docs/cli/build-auth-error/page.mdx
Comment (mlejva) [inline apps/web/src/app/(docs)/docs/cli/build-auth-error/page.mdx:1]: Please create a troubleshooting (eg "Common errors") section in docs and move the page there

File: packages/cli/src/commands/template/buildWithProxy.ts [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+    )
+    const yes = await confirm('Did you complete the guide?')
+
+    if (!yes) {
+      console.log('Please configure the Docker CLI and try again.')
```
Comment (ValentaTomas) [inline packages/cli/src/commands/template/buildWithProxy.ts]:
```suggestion
      console.log('Please follow the workaround steps from https://e2b.dev/docs/cli/build-auth-error and then try again.')
```

File: packages/cli/src/commands/template/buildWithProxy.ts [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+  connectionConfig: e2b.ConnectionConfig,
+  accessToken: string,
+  template: { templateID: string; buildID: string },
+  root: string
+) {
+  if (!userConfig?.dockerProxySet) {
+    console.log(
+      "There was an issue with Docker authentication. Here's a temporary workaround: https://e2b.dev/docs/cli/build-auth-error. Once you've followed the guide, please continue."
+    )
+    const yes = await confirm('Did you complete the guide?')
```
Comment (ValentaTomas) [inline packages/cli/src/commands/template/buildWithProxy.ts]: Have you completed the steps from the https://e2b.dev/docs/cli/build-auth-error workaround guide?

Comment (jakubno) [reply packages/cli/src/commands/template/buildWithProxy.ts]:
```suggestion
    const yes = await confirm('Have you completed the steps from the https://e2b.dev/docs/cli/build-auth-error workaround guide?')
```

File: apps/web/src/app/(docs)/docs/cli/build-auth-error/page.mdx
Comment (jakubno) [reply apps/web/src/app/(docs)/docs/cli/build-auth-error/page.mdx:1]: Added troubleshooting section

**Pull request merged**